pkg/accountmanager/manager.go

```diff
@@ -159,31 +164,56 @@ func (a *AccountManager) RemoveAccount(namespacedName *types.NamespacedName) err
 // restart poller once done.
 func (a *AccountManager) AddResourceFiltersToAccount(accNamespacedName *types.NamespacedName,
 	selectorNamespacedName *types.NamespacedName, selector *crdv1alpha1.CloudEntitySelector, replay bool) (bool, error) {
+	var retError error = nil
+
+	defer func() {
+		// Upon restart, wait for all CES to be added before starting cloud inventory poll.
+		// accountToSelectorMap holds number of CES CRs only in restart case if any CES CR is present in etcd.
+		pendingCes, found := a.accountToNumSelectors[*accNamespacedName]
+		if found {
+			pendingCes--
+			a.accountToNumSelectors[*accNamespacedName] = pendingCes
+		}
+		if !found || pendingCes == 0 {
+			delete(a.accountToNumSelectors, *accNamespacedName)
+			accPoller, exists := a.getAccountPoller(accNamespacedName)
+			if exists {
+				accPoller.restartPoller(accNamespacedName)
+				// wait for polling to complete after restart.
+				retError = accPoller.waitForPollDone(accNamespacedName)
+			}
+		}
+	}()
+
 	cloudProviderType, ok := a.getAccountProviderType(accNamespacedName)
 	if !ok {
-		return true, fmt.Errorf(fmt.Sprintf("failed to add or update selector %v, account %v: "+
-			"provider type not found", selectorNamespacedName, accNamespacedName))
+		retError = fmt.Errorf(fmt.Sprintf("failed to add or update selector %v, account %v: "+
+			"provider type not found", *selectorNamespacedName, *accNamespacedName))
+		return false, retError
```

Comment:
What is the use of creating a new error variable here vs directly returning the error?

Reply:
we want the defer function to get executed too which can alter the return value.